internal/statestorage/redis/playerindices/playerindices.go

```diff
@@ -0,0 +1,273 @@
+//   "indices": [
+//     "pings.us-east",
+//     "modes.ctf",
+//     "modes.teamdm",
+//     "maps.sunsetvalley",
+//     "mmr.rating",
+//   ]
+//
+// For now, OM reads your 'config/matchmaker_config.(json|yaml)' file for a
+// list of indices, which it monitors using the golang module Viper
+// (https://github.com/spf13/viper).
+// In a full deployment, it is expected that you don't manage the config file
+// directly, but instead put the contents of that file into a Kubernetes
+// ConfigMap.  Kubernetes will write those contents to a file inside your
+// running container for you. You can see where and how this is happening by
+// looking at the kubernetes deployment resource definitions in the
+// 'deployments/k8s/' directory.
+//
+// https://github.com/GoogleCloudPlatform/open-match/issues/42 discusses more
+// about how configs are managed in Open Match.
+//
+// You can update the list of indices at run-time if you need to add or remove
+// an index. Changes will affect all indexed players that come in the Frontend
+// API from that point on.
+// NOTE: there are potential edge cases here; see Retrieve() for details.
+
+// Create indices for given player attributes in Redis.
+// TODO: make this quit and not index the player if the context is cancelled.
+func Create(ctx context.Context, rPool *redis.Pool, cfg *viper.Viper, player om_messages.Player) error {
+
+	// Connect to redis
+	redisConn := rPool.Get()
+	defer redisConn.Close()
+
+	iLog := piLog.WithFields(log.Fields{"playerId": player.Id})
+
+	// Get the indices from viper
+	indices, err := Retrieve(cfg)
+	// Get metadata indicies
+	indices = append(indices, MetaIndices...)
+	if err != nil {
+		iLog.Error(err.Error())
+		return err
+	}
+
+	// Start putting this player into the indices in Redis.
+	redisConn.Send("MULTI")
+	// Loop through all attributes we want to index.
+	for _, attribute := range indices {
+		// Default value for all attributes if missing or malformed is the current epoch timestamp.
+		value := time.Now().Unix()
+
+		// If this is a user-defined index, look for it in the input player properties JSON
+		if !strings.HasPrefix(attribute, "OM_METADATA") {
+			// NOTE: This gjson call has issues with JSON keys containing meta characters (dot, slash, etc).
+			//   The regexp.QuoteMeta below gets around those issues, but won't pick up dot-notation keys!
+			//   End result is that you shouldn't use meta characters in your JSON property keys!
+			//v := gjson.Get(player.Properties, regexp.QuoteMeta(attribute))
+			v := gjson.Get(player.Properties, attribute)
+
+			// If this attribue wasn't provided in the JSON, continue to the
+			// next attribute to index.
+			if !v.Exists() {
+				iLog.WithFields(log.Fields{"attribute": attribute, "value": v.Raw}).Debug("Couldn't find index in JSON: ", player.Properties)
+				continue
+			} else if -9223372036854775808 <= v.Int() && v.Int() <= 9223372036854775807 {
+				// value contains a valid unsigned 64-bit integer
```

Comment:
My gjson + golang types is a little rough, but would this be a signed int? I definitely think it makes sense as a signed int64 and not a uint64, so the code is right, but the comment doesn't match.